packages/payment-detection/src/reference-based-detector.ts

```diff
@@ -77,19 +77,22 @@ export default abstract class ReferenceBasedDetector<TPaymentEventParameters>
   public async getBalance(
     request: RequestLogicTypes.IRequest,
   ): Promise<PaymentTypes.IBalanceWithEvents<TPaymentEventParameters>> {
-    if (!request.currency.network) {
-      request.currency.network = 'mainnet';
-    }
+    // TODO
+    // if (!request.currency.network) {
+    //   request.currency.network = 'mainnet';
+    // }
+    const paymentNetwork = request.extensions[this.extensionType];
+    const networkOfPayment = this.getNetworkOfPayment(request.currency, paymentNetwork);
+
     const supportedNetworks = this.extension.supportedNetworks;
-    if (!supportedNetworks.includes(request.currency.network)) {
+    if (!supportedNetworks.includes(networkOfPayment)) {
```

Comment:
We use the name "payment chain" to make the difference between "[RN] payment network" and "payment "currency" network", ok for you to rename such variables `paymentChain` ?